src/network/terrier_server.cpp

```diff
@@ -42,41 +42,43 @@ void TerrierServer::RunServer() {
   // This line is critical to performance for some reason
   evthread_use_pthreads();
 
-  int conn_backlog = common::Settings::CONNECTION_BACKLOG;
-
-  struct sockaddr_in sin;
-  std::memset(&sin, 0, sizeof(sin));
-  sin.sin_family = AF_INET;
-  sin.sin_addr.s_addr = INADDR_ANY;
-  sin.sin_port = htons(port_);
-
-  listen_fd_ = socket(AF_INET, SOCK_STREAM, 0);
-
-  if (listen_fd_ < 0) {
-    NETWORK_LOG_ERROR("Failed to open socket: {}", strerror(errno));
-    throw NETWORK_PROCESS_EXCEPTION("Failed to open socket.");
-  }
-
-  int reuse = 1;
-  setsockopt(listen_fd_, SOL_SOCKET, SO_REUSEADDR, &reuse, sizeof(reuse));
-
-  int retval = bind(listen_fd_, reinterpret_cast<struct sockaddr *>(&sin), sizeof(sin));
-  if (retval < 0) {
-    NETWORK_LOG_ERROR("Failed to bind socket: {}", strerror(errno));
-    throw NETWORK_PROCESS_EXCEPTION("Failed to bind socket.");
-  }
-  retval = listen(listen_fd_, conn_backlog);
-  if (retval < 0) {
-    NETWORK_LOG_ERROR("Failed to create listen socket: {}", strerror(errno));
-    throw NETWORK_PROCESS_EXCEPTION("Failed to create listen socket.");
+  // int conn_backlog = common::Settings::CONNECTION_BACKLOG;
+
+  for (auto &protocol : protocols_) {
```

Comment:
Nit: this logic probably should be broken out into a function. Additionally, this seems to duplicate existing code.